x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/categorization/CategorizeTextAggregationBuilder.java

```diff
@@ -362,6 +315,6 @@ public String getType() {
 
     @Override
     public Version getMinimalSupportedVersion() {
-        return Version.V_7_16_0;
+        return Version.V_8_3_0;
```

Comment:
How does this work when somebody does `categorize_text` from a 7.16 node and it attempts to serialize to a 8.3 node? 

I know this minimal support version prevents us from WRITING to older nodes, but what if an older node is writing to it? 

We may need to add a clause at the top of stream input parsing to make sure the input stream is at least 8.3?

Reply:
This is a good point.

While I was making changes I realised that a newer node _can_ actually use the new algorithm and respond to an older coordinating node in the format that the older node wants.  So I have implemented that.  The merging of categories on the coordinating node might not work brilliantly if different nodes have created their local categories in different ways, but it should be better than nothing.

With the changes of b3fe740f3dac21d207e0f42e1afd8cceab5752e6 I think the communications between old and new nodes should at least be safe, even though the results might be a bit strange sometimes.